testsuite/tests/shape-index/index_aliases.reference

```diff
@@ -0,0 +1,33 @@
+Indexed shapes:
+Resolved_alias: Index_aliases.10 -> Index_aliases.2 -> Index_aliases.1 :
+  G (File "index_aliases.ml", line 29, characters 8-9)
+Resolved_alias: Index_aliases.2 -> Index_aliases.1 :
+  B (File "index_aliases.ml", line 28, characters 11-12)
+Resolved: Index_aliases.7 :
+  C (File "index_aliases.ml", line 26, characters 11-12)
+Resolved_alias: Index_aliases.2 -> Index_aliases.1 :
+  B (File "index_aliases.ml", line 25, characters 14-15)
+Resolved: Index_aliases.5 :
+  F (File "index_aliases.ml", line 25, characters 12-13)
+Resolved: Index_aliases.1 :
+  A (File "index_aliases.ml", line 23, characters 14-15)
+Resolved_alias: Index_aliases.6 -> Index_aliases.5 :
+  F' (File "index_aliases.ml", line 23, characters 11-13)
+Resolved: Index_aliases.5 :
+  F (File "index_aliases.ml", line 22, characters 12-13)
+Resolved: Index_aliases.4 :
+  X (File "index_aliases.ml", line 21, characters 32-33)
```

Comment:
Question: what happens at the merlin to those occurrences that are resolved to functor arguments that are not stored in the declaration index?
(A small feature wish for later: reorder the index printing to be in the uid order).

Reply:
They are ignored by Merlin. The current indexing process does not filter-out usages of definitions that are purely local or not exported by the module. I mentioned it to  @gasche at some point and we arrived at the conclusion that it would probably be premature optimization: since the cmt files already contains the Typedtree, and the index is just a different view over it, it should cost very little in term of space to keep these extraneous occurrences. If at some point we decide to drop the Typedtree from the cmt files, then it might be interested to filter these.

Merlin ignores the index for the current buffer and instead re-index it using the new compiler's iterators while additionally keeping track of local declarations.

Reply:
If you think that would be useful we could at least easily filter the occurrences of "obviously" local definitions (like functor arguments, ` let ... in` definitions etc) by simply checking if the resolved uid is present or not in the declaration index.